### ISSUE TITLE
Write wallaby config in alternative languages (e.g. TypeScript)

### ISSUE THREAD

**Issue by tonyxiao** [wallabyjs/public#1470]
Is it possible to write `wallaby.js` configuration in other languages such as TypeScript? Our entire codebase is in typescript and we try our best to keep it that way (so the compiler and linter and formatter can all do their job)

**Comment by ArtemGovorov** [comment]
Right now it's not possible, so adding the feature request label.

**Comment by marionebl** [comment]
I'd imagine something along the lines of

```js
/**
 * @type {import('wallaby').Config}
 */
const config = {
  autoDetect: true,
  tests: [
    '**/*.test.ts',
    '**/*.test.tsx',
    { pattern: '**/__tests__/**', ignore: true }
  ],
};

module.exports = config;

```

Prior art for this is eslint:

```
/**
 * @type {import('eslint').Linter.Config}
 */
const config = {
  /* ... */
}

module.exports = config;
```

**Comment by ArtemGovorov** [comment]
@marionebl Thanks for sharing the config. We'll investigate adding config types. 

Quick question: is the shared config just an example, or a real config?

```js
/**
 * @type {import('wallaby').Config}
 */
const config = {
  autoDetect: true,
  tests: [
    '**/*.test.ts',
    '**/*.test.tsx',
    { pattern: '**/__tests__/**', ignore: true }
  ],
};

module.exports = config;
```

The reason I'm asking is because it seems like the config may not be required at all (if your Jest config already specifies the right test patterns).

**Comment by smcenlly** [comment]
> due to test files in `__tests__` we know aren't compatible with wallaby

Interested to understand if these are `jest` tests and to know if there's a problem with Wallaby we should be looking at for you?

**Comment by marionebl** [comment]
> Interested to understand if these are jest tests and to know if there's a problem with Wallaby we should be looking at for you?

No, those are playwright tests with a slightly different TypeScript setup - wouldn't expect Wallaby to support those.
